modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -1063,6 +1072,14 @@ def duty_begin_date_is_after_approximate_begin_date?(begin_date, approximate_beg
         date_regex_groups(begin_date) > date_regex_groups(approximate_begin_date)
       end
 
+      def overlapping_confinement_periods?(idx)
+        @ranges.count > 1 ? date_range_overlap?(@ranges[idx - 1], @ranges[idx]) : return
+      end
+
+      def date_range_overlap?(range_one, range_two)
+        range_one.last > range_two.first || range_two.last < range_one.first
```

Comment:
@stiehlrod  Since either of these dates can be null it looks like we will need presence checks on all this data as well.  Image is of the test I just sent in, the first confinement is what broke it with null.

![Screenshot 2024-06-21 at 9 57 58 AM](https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/6fb12079-f8ac-49ef-903e-314fb7301648)

Reply:
@rockwellwindsor-va @FonzMP So our schema allows for null for the approx begin date, and we have a raise exception if it is [blank](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb#L691). It is the same way for approx end date. So should I remove lines 690-697?

Reply:
It does look like 690-697 shouldn't be in there anymore based on the current state.  I think lines 699-703 would also need to be adjusted since the dates are optional.   I think 699 should probably just go away as well (?) and then the check for a valid date (line 700) should only occur if the date is present, so two checks for the begin and end date.  

I believe I was wrong in my previous comment.  The null date triggers the `next` as it should.  It looks like the way `@ranges` is built up may cause an issue.

Reply:
@rockwellwindsor-va thanks! i will remove those here, and then make a ticket to double check for other such examples.

Reply:
I don't think we can remove all that confinement date validation? If I call 526/validate with a null confinement approximate end date, it will now pass validation; however if I submit a 526 with the same request having null confinement date, I get a docker container error for invalid field value. We don't want to allow such requests to generate an error during later processing.

```
2024-06-24 09:51:39.584671 I [47626:puma srv tp 004] Rails -- ClaimsApi :: 526_docker_container :: {"detail":"EVSS DOCKER CONTAINER validate error: {:messages=>[{:key=>\"form526.serviceInformation.confinements[0].confinementEndDate.IsNotValid\", :severity=>\"ERROR\", :text=>\"Invalid value: null\"}]}","claim":"23ef4fa4-ea18-49e9-908a-42fd9d40359d"} :: Location: /Users/matthewchristianson/development/vets-api/modules/claims_api/lib/evss_service/base.rb:93
2024-06-24 09:51:39.586089 I [47626:puma srv tp 004] Rails -- Completed 422 Unprocessable Entity in 13498ms (Allocations: 484818)
```

Here is the serviceInformation portion of my request:
```
"serviceInformation":{
            "servicePeriods":[
                {
                "serviceBranch":"Air Force Reserves",
                "serviceComponent":"Reserves",
                "activeDutyBeginDate": "1990-04-05",
                "activeDutyEndDate": "1998-12-31",
                "seperationLocationCode": null
                },
                {
                "serviceBranch":"Air Force",
                "serviceComponent":"Active",
                "activeDutyBeginDate": "1985-02-05",
                "activeDutyEndDate": "1990-01-02",
                "seperationLocationCode": null
                }
            ],
	        "confinements": [
		        {"approximateBeginDate": "1987-02", "approximateEndDate": null}
                ,
                {"approximateBeginDate": "1990-04-05", "approximateEndDate": "1998-12"}
	        ]
         }
```